miniconf_derive/src/field.rs

```diff
@@ -8,14 +7,21 @@ pub struct StructField {
 
 impl StructField {
     pub fn new(field: syn::Field) -> Self {
-        let attributes: Vec<MiniconfAttribute> = field
-            .attrs
-            .iter()
-            .filter(|attr| attr.path.is_ident("miniconf"))
-            .map(|attr| AttributeParser::new(attr.tokens.clone()).parse())
-            .collect();
+        let mut deferred = false;
 
-        let deferred = attributes.iter().any(|x| *x == MiniconfAttribute::Defer);
+        for attr in field.attrs.iter() {
+            if attr.path().is_ident("miniconf") {
+                attr.parse_nested_meta(|meta| {
+                    if meta.path.is_ident("defer") {
```

Comment:
The existing style, while some code overhead, was intended to make it easy to add new attributes while making it relatively maintainable. This is less code, but makes it harder to track new attributes in the future

Reply:
It's just another `if`, i.e. about five lines.
Same amount as in the old code (c.f. https://github.com/quartiq/miniconf/pull/150/files#diff-1857dc627a1f48fd67d28c8ed2d35e53994fe4b9d71ee181c034cc98a30bd163)